Proxy API requests through the front end.

MancunianSam opened the pull request:
* Add controller endpoints to replace the API endpoints.
* Add a service to call the original API endpoints.
* Add tests for the new controller and service methods.
* Update the javascript to call the front end instead of the API
* Update the jest tests to work with the new fetch calls
* Remove apollo client and the graphql module.

TomJKing commented on app/controllers/UploadController.scala:
```diff
@@ -1,27 +1,56 @@
+    }
+  }
+
+  def saveMetadata(): Action[AnyContent] = secureAction.async { implicit request =>
```
Maybe rename this: `saveClientMetadata`

TomJKing commented on app/controllers/UploadController.scala:
```diff
@@ -62,29 +91,29 @@ class UploadController @Inject()(val controllerComponents: SecurityComponents,
+        reference <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
+      } yield {
+        val uploadStatus: Option[String] = consignmentStatus.flatMap(_.upload)
+        val pageHeading1stHalf = "Upload judgment"
```
Maybe rename these vals:

- `pageHeadingUpload`
- `pageHeadingUploading`

TomJKing commented on app/services/UploadService.scala:
```diff
@@ -0,0 +1,42 @@
+    )
+  }
+
+  def saveMetadata(addFileAndMetadataInput: AddFileAndMetadataInput, token: BearerAccessToken): Future[List[afam.AddFilesAndMetadata]] = {
```
Maybe rename: `saveClientMetadata`

TomJKing commented on test/services/UploadServiceSpec.scala:
```diff
@@ -0,0 +1,73 @@
+  private val graphQlConfig = mock[GraphQLConfiguration]
+  private val token = new BearerAccessToken("some-token")
+
+  "startUpload" should "returns the correct data" in {
```
Reword slightly: `... return the correct data`

TomJKing commented on test/services/UploadServiceSpec.scala:
```diff
@@ -0,0 +1,73 @@
+    response should equal("ok")
+  }
+
+  "saveMetadata" should "returns the correct data" in {
```
Reword slightly: `... return the correct data`

TomJKing commented on app/services/UploadService.scala:
```diff
@@ -0,0 +1,42 @@
+    })
+  }
+
+  def startUpload(startUploadInput: StartUploadInput, token: BearerAccessToken): Future[String] = {
```
Is it possible to have no response here? If so, should an error be thrown?

MancunianSam replied:
The `ApiErrorHandling` object calls `result.data.get` so if data was optional, we'd get a (slightly ugly) error and it also throws a `GraphQlException` if the API populates the errors field. I think that covers everything that could happen. What do you think?

TomJKing reviewed (commented):
Looks good. Some minor comments.

Also when testing locally seeing the following error when trying to upload a judgment. Standard uploads working OK.

![Screenshot from 2022-07-11 07-49-43](https://user-images.githubusercontent.com/51711388/178205429-14c20b52-3b58-4e17-947c-b804e45346be.png)

MancunianSam commented:
I only tested this for a standard user and forgot there's different views for the judgment users. The CSRF input was missing so I've added that in and added in a missing entry to the routes file which was causing an error further on. You can now do a complete judgment export.

TomJKing reviewed (approved):
Looks good. Thanks
